py/torch_tensorrt/dynamo/partitioning/_hierarchical_partitioner.py

```diff
@@ -0,0 +1,574 @@
+        for n in self.module.graph.nodes:
+            n.backend = "None"
+            if n.op in CALLABLE_NODE_OPS:
+                is_supported, backend = self.operator_support.is_node_supported(
+                    submodules, n
+                )
+                if is_supported:
+                    n.backend = backend
+                    self.acc_nodes.add(n)
+
+        if not self.allow_non_tensor:
+            self.reduce_acc_nodes_non_tensor_input()
+            self.reduce_acc_nodes_non_tensor_output()
+
+        return self.acc_nodes
+
+
+@compatibility(is_backward_compatible=False)
+class FxNetSplitterInternalError(Exception):
+    pass
+
+
+def hierarchical_adjacency_partition(
+    gm: torch.fx.GraphModule,
+    verbose: bool = DEBUG,
+    min_block_size: int = MIN_BLOCK_SIZE,
+    torch_executed_ops: Collection[Target] = set(),
+    backend_support_map: Optional[Dict[str, Set[OpOverload]]] = None,
+    backend_priority: Optional[List[str]] = None,
+    require_full_compilation: bool = REQUIRE_FULL_COMPILATION,
+    skip_fusion: bool = False,
+) -> Tuple[torch.fx.GraphModule, BackendOpSupportTester]:
+    """Partition an FX GraphModule with aten ops into submodules using hierarchical partitioning
+    based on backend priority and operator support
+
+    Args:
+        gm: FX GraphModule to partition
+        verbose: Bool representing whether to print operator support
+        min_block_size: Minimum number of operators per TRT-Engine Block
+        backend_support_map: Dictionary mapping backend names to sets of supported operators
+        backend_priority: Ordered list of backend names, from highest to lowest priority
+        require_full_compilation: Require that all computational operators be run in TRT
+        skip_fusion: Skip fusions found by FxNetAccFusionsFinder
+    Returns:
+        torch.fx.GraphModule, BackendOpSupportTester
+    """
+    # Ensure graph is clean prior to partitioning
+    gm.graph.eliminate_dead_code()
+    gm.graph.lint()
+    gm.recompile()
+
+    # Default backend support map if none provided
+    if backend_support_map is None:
+        backend_support_map = {
+            "tensorrt": set(DYNAMO_ATEN_CONVERTERS.keys()),
+            "inductor": set(),
+        }
+
+    # Default backend priority if none provided
+    if backend_priority is None:
+        backend_priority = ["tensorrt", "inductor"]
+
+    # Construct BackendOpSupportTester
+    supported_ops = BackendOpSupportTester(
+        backend_support_map=backend_support_map,
+        backend_priority=backend_priority,
+        torch_executed_ops=torch_executed_ops,
+    )
+    partitioner = HierarchicalAdjacencyPartitioner(
+        gm,
+        supported_ops,
+        backend_support_map=backend_support_map,
+        backend_priority=backend_priority,
+        min_block_size=min_block_size,
+        require_full_compilation=require_full_compilation,
+        skip_fusion=skip_fusion,
+    )
+
```

Comment:
Do you think copying all this code is necessary ? or inheriting only some components would be sufficient ?